include/envoy/http/header_map.h

```diff
@@ -315,6 +315,7 @@ class HeaderEntry {
   HEADER_FUNC(GrpcAcceptEncoding)                                                                  \
   HEADER_FUNC(GrpcMessage)                                                                         \
   HEADER_FUNC(GrpcStatus)                                                                          \
+  HEADER_FUNC(GrpcStatusDetailsBin)                                                                \
```

Comment:
perf nit: given that this header is only used in error cases, my preference would be to not look it up via O(1) and just have it looked up by name? @lizan any thoughts here?

Reply:
Look up by name works for me, I don't have strong opinion. The downside of having headers here is more memory usage per request?

Reply:
Yes, basically we have to allocate space and zero out the memory for every header map. It's not free. With that said, we don't really know what the cost is because we don't have a microbenchmark. @cpakulski is looking at this in the context of https://github.com/envoyproxy/envoy/issues/4815. cc @jmarantz @oschaaf

Reply:
cc @asraa as well who is looking at refactoring header-map for improved fuzzing.

Reply:
> looking at refactoring header-map for improved fuzzing

@asraa can we open a design proposal on that separately to discuss before doing a bunch of work? I think that would be optimal.